cranelift-codegen/src/binemit/mod.rs

```diff
@@ -127,6 +127,9 @@ pub trait CodeSink {
     /// Add a relocation referencing an external symbol plus the addend at the current offset.
     fn reloc_external(&mut self, _: Reloc, _: &ExternalName, _: Addend);
 
+    /// Add a relocation referencing a jump table.
+    fn reloc_constant(&mut self, _: Reloc, _: ConstantOffset);
```

Comment:
Here we tell embedders that there is also the possibility that a constant (located at some offset in a constant pool) will need to be relocated. `ConstantOffset` is a re-typed `u32`. One issue I see with this is that currently all of the offsets are calculated at a function level: e.g. if a function has one constant then its `ConstantOffset` will be `0`, but if there are three 16-byte constants in the entire code (each in different functions), then the embedder would be told to relocate the constant at offset `0` three times...

Reply:
Every function has its own `CodeSink`. Cranelift is made for separate compilation of functions, so you build one function and compile it and then the next.